### PR TITLE
Update edgecmd-commands.md

### PR DESCRIPTION
**Pull request by TinaUll**
Hi, I've added the data discovery config to the edgecmd reference. Please let me know if anything is missing or incorrect. Thank you!

### CODE REVIEW

File: V1.2/Reference/edgecmd-commands.md [hunk 1]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
 `edgecmd get Version` | Get version information.
+
+## Data discovery configuration
+
+EdgeCmd command| Description | Examples |
+---------------|-------------|----------|
+`edgecmd add discoveries -cid <componentId>` | Add data source discoveries to specified component | `edgecmd add discoveries -cid Mqtt1`|
+`edgecmd get discoveries -cid <componentId>` | Get active data source discoveries for specified component | `edgecmd get discoveries -cid Mqtt1`|
+| `edgecmd get discoveries -cid <componentId> -id <discoveryId> -result [-csv] [-file <filename>]` | Get the result of specified discovery for specified component | `edgecmd get discoveries -cid Mqtt1 -id 1 -result` |
+| `edgecmd add discoveries -cid <componentId> -query "<queryFilter>" -id <discoveryId>` | Search all items that have the query filter string in specified discovery of specified component | `edgecmd add discoveries -cid Mqtt1 -query "+/+/Device90" -id 1` |
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:213]: This starts a new discovery operation with query "+/+/Device90" and Id "1"

File: V1.2/Reference/edgecmd-commands.md [hunk 2]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
 ---------------|-------------|
 `edgecmd get Diagnostics` | Get the diagnostics configuration. |
 `edgecmd get Version` | Get version information.
+
+## Data discovery configuration
+
+EdgeCmd command| Description | Examples |
+---------------|-------------|----------|
+`edgecmd add discoveries -cid <componentId>` | Add data source discoveries to specified component | `edgecmd add discoveries -cid Mqtt1`|
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:210]: This starts a new discovery operation and returns status object containing generated Id of the discovery

File: V1.2/Reference/edgecmd-commands.md [hunk 3]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+`edgecmd get discoveries -cid <componentId>` | Get active data source discoveries for specified component | `edgecmd get discoveries -cid Mqtt1`|
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:211]: This present you with ative as well as completed discovery status information. (array of discovery operation statuses)

File: V1.2/Reference/edgecmd-commands.md [hunk 4]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd get discoveries -cid <componentId> -id <discoveryId> -result [-csv] [-file <filename>]` | Get the result of specified discovery for specified component | `edgecmd get discoveries -cid Mqtt1 -id 1 -result` |
+| `edgecmd add discoveries -cid <componentId> -query "<queryFilter>" -id <discoveryId>` | Search all items that have the query filter string in specified discovery of specified component | `edgecmd add discoveries -cid Mqtt1 -query "+/+/Device90" -id 1` |
+| `edgecmd get discoveries -cid <componentId> -id <discoveryId> -result -query diff=<discoveryId> [-csv]` | Search the differences between two specified discoveries of specified component | `edgecmd get discoveries -cid Mqtt1 -id 1 -result -query diff=2` |
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:214]: Would be Get differences better?

File: V1.2/Reference/edgecmd-commands.md [hunk 5]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd add dataselection -cid <componentId> -select -query discoveryId=<discoveryId>` | Add specified discovery to data selection and select automatically | `edgecmd add dataselection -cid Mqtt1 -select -query discoveryId=1` |
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:215]: Add specified discovery result to data selection and set all the items to selected - or something along those lines?

File: V1.2/Reference/edgecmd-commands.md [hunk 6]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd get dataselection -cid <componentId> [-csv]` | Get discoveries added to data selection | `edgecmd get dataselection -cid Mqtt1` |
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:216]: This is part of dataselection - not really related to discovery but maybe it's useful to have it here

File: V1.2/Reference/edgecmd-commands.md [hunk 7]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd get dataselection -cid <componentId> -query diff=<discoveryId> [-csv]` | Compare data selection against a discovery | `edgecmd get dataselection -cid Mqtt1 -query diff=2` |
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:217]: against a discovery result

File: V1.2/Reference/edgecmd-commands.md [hunk 8]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd  remove discoveries -cid <componentId> -id <discoveryId> -result [-y]` | Remove only the result of the specified discovery from the specified component | `edgecmd  remove discoveries -cid Mqtt1 -id 2 -result -y`|
+| `edgecmd  remove discoveries -cid <componentId> -id <discoveryId> [-y]` | Remove the specified discovery from the specified component | `edgecmd  remove discoveries -cid Mqtt1 -id 2 -y`|
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:219]: Remove all discovery states as well as results.

File: V1.2/Reference/edgecmd-commands.md [hunk 9]
```diff
@@ -202,3 +202,18 @@ EdgeCmd command| Description |
...
+| `edgecmd get discoveries -cid <componentId> -id <discoveryId> -result -query diff=<discoveryId> [-csv]` | Search the differences between two specified discoveries of specified component | `edgecmd get discoveries -cid Mqtt1 -id 1 -result -query diff=2` |
+| `edgecmd add dataselection -cid <componentId> -select -query discoveryId=<discoveryId>` | Add specified discovery to data selection and select automatically | `edgecmd add dataselection -cid Mqtt1 -select -query discoveryId=1` |
+| `edgecmd get dataselection -cid <componentId> [-csv]` | Get discoveries added to data selection | `edgecmd get dataselection -cid Mqtt1` |
+| `edgecmd get dataselection -cid <componentId> -query diff=<discoveryId> [-csv]` | Compare data selection against a discovery | `edgecmd get dataselection -cid Mqtt1 -query diff=2` |
+| `edgecmd  remove discoveries -cid <componentId> -id <discoveryId> -result [-y]` | Remove only the result of the specified discovery from the specified component | `edgecmd  remove discoveries -cid Mqtt1 -id 2 -result -y`|
```
Comment (zryska) [inline V1.2/Reference/edgecmd-commands.md:218]: It also cancels the discovery if it's still running.

**Review by zryska — CHANGES_REQUESTED** [review]
Some of the sentenses could be rephrased a bit - please see my comments. Thank you!

**Review by rgralakosi — COMMENTED** [review]
The changes look good!

**Pull request merged**